Core: Remove `storyStoreV7` feature flag

ndelangen opened the pull request:
Relates to https://github.com/storybookjs/storybook/issues/25117

## What I did

Step 2 of https://github.com/storybookjs/storybook/pull/24655

## Checklist for Contributors

### Testing

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to communicate how to test your changes -->

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

<!-- Please include the steps to test your changes here. For example:

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Access X story

-->

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

tmeasday committed:
Initial run-through deleting all ssv6 code

tmeasday committed:
Convert all Preview Web code to async/await

tmeasday committed:
Small fixups

tmeasday committed:
Remove public APIs from packages

tmeasday committed:
Drop unused vars

tmeasday committed:
Remove unused var

tmeasday committed:
Drop `buildStoriesJson` flag

tmeasday committed:
Remove unused imports

ndelangen committed:
remove all references of storyStoreV7

ndelangen committed:
delete unused files

ndelangen committed:
remove unused files

ndelangen committed:
cleanup

ndelangen committed:
disable broken tests

ndelangen committed:
cleanup

ndelangen committed:
Merge branch 'tom/remove-ssv6' into norbert/remove-storystorev7

ndelangen committed:
Merge branch 'release-8-0' into norbert/remove-storystorev7

ndelangen committed:
fix

ndelangen committed:
fix

ndelangen committed:
Merge branch 'norbert/remove-storiesof' into norbert/remove-storystorev7

ndelangen committed:
cleanup

ndelangen committed:
Merge branch 'norbert/remove-storiesof' into norbert/remove-storystorev7

ndelangen committed:
make it work

ndelangen committed:
Merge branch 'release-8-0' into norbert/remove-storystorev7

ndelangen committed:
update exports file

ndelangen committed:
Merge branch 'next' into norbert/remove-storystorev7

ndelangen committed:
fix

valentinpalkovic committed:
Merge branch 'next' into norbert/remove-storystorev7

ndelangen committed:
Merge branch 'next' into norbert/remove-storystorev7

ndelangen commented:
@SocketSecurity ignore @vitejs/plugin-vue@4.4.0
@SocketSecurity ignore @babel/preset-modules@0.1.6
@SocketSecurity ignore piscina@4.0.0
@SocketSecurity ignore loupe@2.3.7

ndelangen committed:
fixes

kylegach commented on docs/api/main-config-features.md:
```diff
@@ -107,26 +67,4 @@ Filter args with a "target" on the type from the render function.
-
-<CodeSnippets
-  paths={[
-    'common/main-config-features-story-store-v7.js.mdx',
```
Doing a Find in project for this snippet filename also yields these areas to also be removed:

- https://github.com/storybookjs/storybook/blob/next/docs/configure/index.md?plain=1#L57-L58
- https://github.com/storybookjs/storybook/blob/next/docs/configure/index.md?plain=1#L134-L154
- https://github.com/storybookjs/storybook/blob/next/docs/configure/webpack.md?plain=1#L54-L68

There's also this:

- https://github.com/storybookjs/storybook/blob/next/docs/migration-guide.md?plain=1#L123-L136

But the better move there would be updating the entire guide to be `8.0`-focused, which should probably happen in another PR.

-----

`storyStoreV7` and `buildStoriesJson` also have multiple additional instances throughout the docs that should be removed.

valentinpalkovic replied:
cc @valentinpalkovic

ndelangen committed:
Merge branch 'next' into norbert/remove-storystorev7

ndelangen committed:
Merge branch 'next' into norbert/remove-storystorev7

yannbf commented:
Please add migration notes that cover all breaking changes, especially when it comes to APIs which could break addons 🙏 

These two issues might give a bit of light, but there is definitely much more to cover:
https://github.com/storybookjs/storybook/issues/25367
https://github.com/storybookjs/storybook/issues/25365

ndelangen committed:
Merge branch 'next' into norbert/remove-storystorev7

ndelangen committed:
remove docs references to storyStoreV7

ndelangen committed:
remove more

ndelangen committed:
upgrade to latest icons

ndelangen committed:
upgrades

ndelangen committed:
fix typings

ndelangen committed:
fix snapshots

ndelangen committed:
fix typings

ndelangen committed:
Merge branch 'next' into norbert/remove-storystorev7

shilman commented on code/lib/preview-api/src/modules/preview-web/Preview.tsx:
```diff
@@ -185,15 +163,11 @@ export class Preview<TRenderer extends Renderer> {
 
   // If initialization gets as far as the story index, this function runs.
-  initializeWithStoryIndex(storyIndex: StoryIndex): PromiseLike<void> {
+  initializeWithStoryIndex(storyIndex: StoryIndex): void {
```
Shouldn't this be async?

ndelangen replied:
Let's just make it `Promise<void>`

shilman commented on code/lib/preview-api/src/modules/preview-web/PreviewWeb.integration.test.ts:
```diff
@@ -76,7 +73,7 @@ beforeEach(() => {
 });
 
-describe('PreviewWeb', () => {
+describe.skip('PreviewWeb', () => {
```
What's the plan here? @tmeasday to address in follow-up PR?

shilman commented on code/lib/preview-api/src/modules/preview-web/PreviewWeb.test.ts:
```diff
@@ -142,7 +138,7 @@ beforeEach(() => {
 });
 
-describe('PreviewWeb', () => {
+describe.skip('PreviewWeb', () => {
```
Same q

shilman commented on docs/configure/index.md:
```diff
@@ -52,10 +52,9 @@ This configuration file is a [preset](../addons/addon-types.md) and, as such, ha
-| `buildStoriesJson`    | Generates `index.json` and `stories.json` files to help story loading with the on-demand mode (defaults to `true` when `storyStoreV7` is `true`) <br/> `features: { buildStoriesJson: true }` |
+| Configuration element | Description                                                                                                                                                     |
+| --------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `buildStoriesJson`    | Generates `index.json` and `stories.json` files to help story loading with the on-demand mode (defaults to `true`) <br/> `features: { buildStoriesJson: true }` |
```
Doesn't this PR get rid of `buildStoriesJson`?

shilman reviewed (commented):
Looks great. Left a few minor comments!

valentinpalkovic committed:
Update yarn.lock

valentinpalkovic committed:
Downgrade markdown-to-jsx

The following issue was introduced in 7.4: https://github.com/quantizor/markdown-to-jsx/issues/535
Downgrading to 7.3.2 will fix it

valentinpalkovic committed:
Remove buildStoriesJson references in docs

valentinpalkovic committed:
Remove unnecessary await
